extensions/amp-ad-exit/0.1/amp-ad-exit.js

```diff
@@ -100,18 +119,67 @@ export class AmpAdExit extends AMP.BaseElement {
       'CLICK_Y': true,
     };
     if (target.vars) {
-      for (const customVar in target.vars) {
-        if (customVar[0] == '_') {
-          vars[customVar] = () =>
-              args[customVar] == undefined ?
-                target.vars[customVar].defaultValue : args[customVar];
-          whitelist[customVar] = true;
+      for (const customVarName in target.vars) {
+        if (customVarName[0] == '_') {
+          const customVar =
+              /** @type {./config.Variable} */ (target.vars[customVarName]);
+          if (customVar) {
+            /*
+              Example:
+              The amp-ad-exit target has a variable representing the
+               priority of something, which is defined as follows:
+               "vars": {
+                 "_pty": {
+                   "defaultValue": "unknown",
+                   "iframeTransportSignal":
+                      "IFRAME_TRANSPORT_SIGNAL(vendorXYZ, priority)"
+                 },
+                 ...
+               }
+               The cross-domain iframe of vendorXYZ has sent the
+               following response for the creative:
+                 { priority: medium, category: W }
+               This is just example data. The keys/values in that object can
+               be any strings.
+               The code below will create substitutionFunctions['_pty'],
+               which in this example will return "medium".
+             */
+            substitutionFunctions[customVarName] = () => {
+              if (customVar.hasOwnProperty('iframeTransportSignal')) {
```

Comment:
url-replacement-impl already supports function var.
you should be able to introduce new function by doing 
```
Services.urlReplacementsForDoc(this.getAmpDoc()).getVariableSource().setAsync('IFRAME_TRANSPORT_SIGNAL', resolver);
```

Reply:
Thanks, @lannka. I remember you said there was a better way, and was planning to ask for details but you beat me to it :)

Reply:
Fixed.